scipy/_lib/_array_api.py

```diff
@@ -475,3 +476,155 @@ def xp_sign(x, xp=None):
     sign = xp.where(x < 0, -one, sign)
     sign = xp.where(x == 0, 0*one, sign)
     return sign
+
+
+def xp_add_reduced_axes(res, axis, initial_shape, *, xp=None):
```

Comment:
Could you add a note on why this is needed? Is it temporary, why can't `xp.add` not be used, etc.?

Reply:
Type annotations and consistency with other functions in this file would be useful too (at least if you expect this function to stay around for a while).

`res` should preferably be positional-only.

Reply:
Perhaps a better name would have been `xp_replace_reduced_axes` or `xp_keepdims`: it adds back axes that have been reduced away. When there are other comments to respond to, I'll just move the logic back into `xp_mean`, though, since I'm not sure if it will be used elsewhere. I can also make the first `xp_mean` argument positional-only.